src/dmd/typesem.d

```diff
@@ -1997,13 +1997,12 @@ RootObject compileTypeMixin(TypeMixin tm, Loc loc, Scope* sc)
     const len = buf.length;
     buf.writeByte(0);
     const str = buf.extractSlice()[0 .. len];
-    scope diagnosticReporter = new StderrDiagnosticReporter(global.params.useDeprecated);
-    scope p = new Parser!ASTCodegen(loc, sc._module, str, false, diagnosticReporter);
+    scope p = new Parser!ASTCodegen(loc, sc._module, str, false);
     p.nextToken();
     //printf("p.loc.linnum = %d\n", p.loc.linnum);
 
     auto o = p.parseTypeOrAssignExp(TOK.endOfFile);
-    if (p.errors)
+    if (errors != global.errors)
```

Comment:
The `p.errors` checks have been slightly wrong because they ignore warnings emitted as errors. There are only two warnings shown by parsing and lexing, though.

Reply:
Do you want to make that change together with refactorings?

Reply:
Well, it's more or less a revert of the respective change in the introducing commit. I'm not sure it makes much sense to fix it in the DiagnosticReporter just to rip it out afterwards.